contracts/core/EntryPoint.sol

```diff
@@ -770,7 +770,7 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
     function getMemoryBytesFromOffset(
         uint256 offset
     ) internal pure returns (bytes memory data) {
-        assembly {
+        assembly ("memory-safe") {
```

Comment:
Like, this function is not actually safe, right? It depends on passing it correct `offset`, but this is not immediately obvious, don't you think?
I can see that currently `getMemoryBytesFromOffset` is only used with `context` which is never overwritten, but I think at least the comments and the name of the function should indicate you can shoot yourself in the foot if you use it.
What do you think?
<img width="717" alt="Screenshot 2024-01-08 at 16 59 34" src="https://github.com/eth-infinitism/account-abstraction/assets/40541447/68ad9a17-8ffb-4e72-9efc-035a5b8bab30">